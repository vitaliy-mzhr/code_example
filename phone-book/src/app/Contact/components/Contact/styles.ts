import styled from 'styled-components';

const ContactStyles = styled.div`
  margin: .5rem 0;
  
  .card-action {
    width: 100%;
  }
  .phones-container {
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    align-items: flex-start;
    flex-wrap: nowrap;
    padding: .5rem .5rem .5rem 1rem;
    
    .phone {
      display: flex;
      justify-content: flex-start;
      align-items: center;
      text-decoration: none;
      color: ${props => props.theme.colors.mainColor}
    }
  }
  .phones-header {
    padding: .5rem 1.5rem;
    font-weight: 600;
  }
  .buttons-container {
    position: relative;
    display: flex;
    justify-content: flex-end;
    align-items: center;
  }
  .contact-content {
    display: none;
    padding: 0;
    &.expanded {
      display: block;
    }
    &:last-child {
      padding-bottom: 0;
    }
  }
`;

export default ContactStyles;
