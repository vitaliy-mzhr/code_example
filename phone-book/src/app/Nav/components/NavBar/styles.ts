import styled from 'styled-components';

const NavBarStyles = styled.div`
   width: 100%;
   .app-bar {
    min-height: ${props => props.theme.dimensions.navMinHeightRem}rem;
   }
   .back-button {
      color: ${props => props.theme.colors.whiteColor};
   }
   .home-button {
    margin-left: -12px;
    margin-right: 20px;
    color: ${props => props.theme.colors.whiteColor};
   }
   .search-button {
    color: ${props => props.theme.colors.whiteColor};
   }
   .flex-grow {
    flex-grow: 1;
   }
   .search-container {
    position: relative;
    border-radius: ${props => props.theme.borders.mainBorderRadiusPx}px;
    margin-left: 0;
    background-color: ${props => props.theme.colors.navSearchBackground};
    //width: 100%;
    &:hover {
      background-color: ${props => props.theme.colors.navSearchBackgroundHover};
    }
   }
   .search-icon {
    height: 100%;
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    pointer-events: none;
    width: 4rem;
   }
   .search-input {
    color: inherit;
    padding: .1rem .1rem .1rem 4.1rem;
    input {
      width: 10rem;
    }
    &.add-transition {
      input {
        width: 10rem;
        &:focus {
        width: 15rem;
      }
        transition: width .15s ease-in;
      }
    }
   }
   
   @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {
   .app-bar {
    padding: 0 .7rem;
    min-height: ${props => props.theme.dimensions.navMinHeightRem * 0.9}rem;
   }
    .title {
      display: none;
    }
    .search-input {
      input {
        width: 7rem;
     }
     &.add-transition {
      input {
        width: 7rem;
        &:focus {
        width: 10rem;
      }
     }
     }
    }
  }
`;

export default NavBarStyles;
