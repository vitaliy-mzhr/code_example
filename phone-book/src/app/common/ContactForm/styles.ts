import styled from 'styled-components';

const ContactFormStyles = styled.div`
  h3 {
    text-align: center;
  }
  .form-container {
    margin: 0 auto;
    padding-top: 2rem;
    max-width: 20rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  .buttons-container {
    display: flex;
    justify-content: center;
    padding: 1rem;
    button {
      margin: 0 .25rem;
    }
  }
`;

export default ContactFormStyles;
