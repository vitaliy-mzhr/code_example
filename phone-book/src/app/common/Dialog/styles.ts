import styled from 'styled-components';

const AlertDialogStyles = styled.div`
  min-width: 14rem;
  .dialog-title {
    h2 {
      text-align: center;
    }
  }
  .buttons-container {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
  }
  
  .confirm-button-icon {
    color: ${props => props.theme.colors.greenColor};
  }
  .cancel-button-icon {
    color: ${props => props.theme.colors.errorColor};
  }
`;

export default AlertDialogStyles;
