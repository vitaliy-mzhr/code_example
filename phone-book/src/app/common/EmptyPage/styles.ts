import styled from 'styled-components';

const EmptyPageStyles = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  .page-message-container {
    margin-top: 2rem;
    padding: 2rem;
}
`;

export default EmptyPageStyles;
