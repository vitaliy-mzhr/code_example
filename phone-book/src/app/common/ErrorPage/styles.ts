import styled from 'styled-components';

const ErrorPageStyles = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  .page-message-container {
    margin-top: 2rem;
    padding: 2rem;
    color: ${props => props.theme.colors.errorColor};
}
`;

export default ErrorPageStyles;
