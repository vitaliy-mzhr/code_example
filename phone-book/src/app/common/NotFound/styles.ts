import styled from 'styled-components';

const NotFoundStyles = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  
  a {
    text-decoration: none;
  }
`;

export default NotFoundStyles;
