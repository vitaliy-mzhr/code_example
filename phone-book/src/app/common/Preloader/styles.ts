import styled from 'styled-components';

const PreloaderStyles = styled.div`
  position: absolute;
  width: 100%;
  height: calc(100% - ${props => props.theme.dimensions.navMinHeightRem}rem);
  display: flex;
  justify-content: center;
  align-items: center;
  
  @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {
   height: calc(100% - ${props => props.theme.dimensions.navMinHeightRem * 0.9}rem);
  }
`;

export default PreloaderStyles;