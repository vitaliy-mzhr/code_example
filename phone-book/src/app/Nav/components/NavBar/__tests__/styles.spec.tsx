import * as React from 'react';
import NavBarStyles from '../styles';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { mainTheme } from '../../../../../styles/themes';

it('NavBarStyles applied correctly', () => {
  expect(shallow(<NavBarStyles theme={mainTheme} />)).toMatchSnapshot();
});