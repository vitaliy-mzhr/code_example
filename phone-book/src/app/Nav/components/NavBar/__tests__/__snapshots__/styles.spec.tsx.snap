// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeStyles applied correctly 1`] = `
.c0 {
  width: 100%;
}

.c0 .app-bar {
  min-height: 4.5rem;
}

.c0 .back-button {
  color: #fff;
}

.c0 .home-button {
  margin-left: -12px;
  margin-right: 20px;
  color: #fff;
}

.c0 .search-button {
  color: #fff;
}

.c0 .flex-grow {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .search-container {
  position: relative;
  border-radius: 4px;
  margin-left: 0;
  background-color: rgba(256,256,256,.15);
}

.c0 .search-container:hover {
  background-color: rgba(256,256,256,.25);
}

.c0 .search-icon {
  height: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  width: 4rem;
}

.c0 .search-input {
  color: inherit;
  padding: .1rem .1rem .1rem 4.1rem;
}

.c0 .search-input input {
  width: 10rem;
}

.c0 .search-input.add-transition input {
  width: 10rem;
  -webkit-transition: width .15s ease-in;
  transition: width .15s ease-in;
}

.c0 .search-input.add-transition input:focus {
  width: 15rem;
}

@media all and (max-width:768px) {
  .c0 .app-bar {
    padding: 0 .7rem;
    min-height: 4.05rem;
  }

  .c0 .title {
    display: none;
  }

  .c0 .search-input input {
    width: 7rem;
  }

  .c0 .search-input.add-transition input {
    width: 7rem;
  }

  .c0 .search-input.add-transition input:focus {
    width: 10rem;
  }
}

<div
  className="c0"
/>
`;

exports[`NavBarStyles applied correctly 1`] = `
.c0 {
  width: 100%;
}

.c0 .app-bar {
  min-height: 4.5rem;
}

.c0 .back-button {
  color: #fff;
}

.c0 .home-button {
  margin-left: -12px;
  margin-right: 20px;
  color: #fff;
}

.c0 .search-button {
  color: #fff;
}

.c0 .flex-grow {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .search-container {
  position: relative;
  border-radius: 4px;
  margin-left: 0;
  background-color: rgba(256,256,256,.15);
}

.c0 .search-container:hover {
  background-color: rgba(256,256,256,.25);
}

.c0 .search-icon {
  height: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  width: 4rem;
}

.c0 .search-input {
  color: inherit;
  padding: .1rem .1rem .1rem 4.1rem;
}

.c0 .search-input input {
  width: 10rem;
}

.c0 .search-input.add-transition input {
  width: 10rem;
  -webkit-transition: width .15s ease-in;
  transition: width .15s ease-in;
}

.c0 .search-input.add-transition input:focus {
  width: 15rem;
}

@media all and (max-width:768px) {
  .c0 .app-bar {
    padding: 0 .7rem;
    min-height: 4.05rem;
  }

  .c0 .title {
    display: none;
  }

  .c0 .search-input input {
    width: 7rem;
  }

  .c0 .search-input.add-transition input {
    width: 7rem;
  }

  .c0 .search-input.add-transition input:focus {
    width: 10rem;
  }
}

<div
  className="c0"
/>
`;
